File: backend/src/main/java/info/gamewise/lor/videos/service/AppSettingsService.java
package info.gamewise.lor.videos.service;

import info.gamewise.lor.videos.port.in.GetAppSettingsUseCase;
import info.gamewise.lor.videos.scheduler.YouTubeVideosScheduler;
import org.springframework.stereotype.Service;

@Service
class AppSettingsService implements GetAppSettingsUseCase {

    @Override
    public AppSettings getAppSettings() {
        return new AppSettings(YouTubeVideosScheduler.updatedAt());
    }

}

File: backend/src/main/kotlin/info/gamewise/lor/videos/service/AppSettingsService.kt
package info.gamewise.lor.videos.service

import info.gamewise.lor.videos.port.`in`.AppSettings
import info.gamewise.lor.videos.port.`in`.GetAppSettingsUseCase
import info.gamewise.lor.videos.scheduler.YouTubeVideosScheduler.Companion.updatedAt
import org.springframework.stereotype.Service

@Service
internal class AppSettingsService : GetAppSettingsUseCase {

    override fun getAppSettings(): AppSettings {
        return AppSettings(updatedAt())
    }

}